includes/class-wc-gateway-payfast-blocks-support.php

```diff
@@ -0,0 +1,90 @@
+<?php
+use Automattic\WooCommerce\Blocks\Assets\Api;
+use Automattic\WooCommerce\Blocks\Payments\Integrations\AbstractPaymentMethodType;
+
+/**
+ * PayFast payment method integration
+ *
+ * @since 1.5.0
+ */
+final class WC_PayFast_Blocks_Support extends AbstractPaymentMethodType {
+	/**
+	 * Name of the payment method.
+	 *
+	 * @var string
+	 */
+	protected $name = 'payfast';
+
+	/**
+	 * Initializes the payment method type.
+	 */
+	public function initialize() {
+		$this->settings = get_option( 'woocommerce_payfast_settings', [] );
+	}
+
+	/**
+	 * Returns if this payment method should be active. If false, the scripts will not be enqueued.
+	 *
+	 * @return boolean
+	 */
+	public function is_active() {
+		return filter_var( $this->get_setting( 'enabled', false ), FILTER_VALIDATE_BOOLEAN );
+	}
+
+	/**
+	 * Returns an array of scripts/handles to be registered for this payment method.
+	 *
+	 * @return array
+	 */
+	public function get_payment_method_script_handles() {
+		$asset_path   = WC_GATEWAY_PAYFAST_PATH . '/build/index.asset.php';
+		$version      = WC_GATEWAY_PAYFAST_VERSION;
+		$dependencies = [];
+		if ( file_exists( $asset_path ) ) {
+			$asset        = require $asset_path;
+			$version      = is_array( $asset ) && isset( $asset['version'] )
+				? $asset['version']
+				: $version;
+			$dependencies = is_array( $asset ) && isset( $asset['dependencies'] )
+				? $asset['dependencies']
+				: $dependencies;
+		}
+		wp_register_script(
+			'wc-payfast-blocks-integration',
+			WC_GATEWAY_PAYFAST_URL . '/build/index.js',
+			$dependencies,
+			$version,
+			true
+		);
+		wp_set_script_translations(
+			'wc-payfast-blocks-integration',
+			'woocommerce-gateway-payfast',
+			WC_GATEWAY_PAYFAST_PATH . '/languages'
+		);
```

Comment:
I don't think these translation files will have translations from `wp.i18n` strings currently unless those translation files are included in the provided languages folder here. This is outside the scope of this PR though because it's a problem general to using `wp.i18n` in wp plugins not hosted on wp.org.

However, if this plugin _is_ hosted on wp.org then we'll want to not provide a third argument here (so the default path for translation files is used).

Reply:
Aaah, I'm a bit confused by this. The extension is indeed [in .org](https://wordpress.org/plugins/woocommerce-payfast-gateway/) but it doesn't seem to be [prepared for localization](https://translate.wordpress.org/projects/wp-plugins/woocommerce-payfast-gateway/), which is probably out of the scope of this PR. I'm not sure what's the best way to proceed with this, in 5f91c8243aa9a5c5c753d8daf1df6a214f5f3867 I left `wp_set_script_translations()` but removed the third argument. So, if I'm understanding this correctly, when localization is fixed, it will pick the translations automatically, is that right?

Reply:
AFAIK yes. Translations would be downloaded from wp.org to the default location and this would ensure they are loaded from that location if present.